### ISSUE TITLE
Only provider should be able to update the connector url 

### ISSUE THREAD

**Issue by dhiren-singh-007** [eclipse-tractusx/portal-backend#1085]
### Current Behavior
Currently in portal backend there is a validation that only host can update the connector url which restricts the provider to update the connector url.

### Expected Behavior
Actually provider is the one who provides the connector so only provider should be able to update the connector url.

### Steps To Reproduce
Provider : ABC
Host : XYZ
1. Create a managed connector for company "XYZ"
2. Update the connector url which is created for "Host Company" , it will throw and error "Company ABC is not the connectors host company"
3. Now login form XYZ company and update the managed connector, now it will allow the update.

**Comment by ntruchsess** [comment]
@dhiren-singh-007 @MaximilianHauer : I think the behavior should depend on the ConnectorType:
* if it's ConnectorTypeId.CONNECTOR_AS_A_SERVICE the provider should be able to update
* if it's ConnectorTypeId.COMPANY_CONNECTOR the host should be able to update

**Comment by dhiren-singh-007** [comment]
@ntruchsess : I think if COMPANY_CONNECTOR type can have a different provider and host then i think it make sense to allow host to update the connector.
But if COMPANY_CONNECTOR will always have provider and host as same id then i think provider validation should be enough.
cc @MaximilianHauer

**Comment by ntruchsess** [comment]
@dhiren-singh-007 : you are right - ConnectorTypeId.COMPANY_CONNECTOR typed connectors are being created with host and provider being set to the same id ( https://github.com/eclipse-tractusx/portal-backend/blob/3a07d788937dc905d9939307f571fa000b535145/src/administration/Administration.Service/BusinessLogic/ConnectorsBusinessLogic.cs#L115 ). It's therefore sufficient (and also safe) to just check the providerid.
cc @MaximilianHauer

**Comment by gopalreddy-bmw** [comment]
Hi @MaximilianHauer 

I tested this issue it's working fine. Can you close this ticket.

Attaching screenshots for references 
<img width="956" alt="Image" src="https://github.com/user-attachments/assets/08723c83-841c-418d-ba45-b25404c74e7f" />
<img width="959" alt="Image" src="https://github.com/user-attachments/assets/e39d7914-8567-4b97-95fd-2ba0a520920d" />
